sql/expression/function/time.go

```diff
@@ -1247,3 +1249,86 @@ func (c CurrTimestamp) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 func (c CurrTimestamp) WithChildren(children ...sql.Expression) (sql.Expression, error) {
 	return NoArgFuncWithChildren(c, children)
 }
+
+const maxCurrentTimestampPrecision = 6
+
+type CurrentTimestamp struct {
+	expression.UnaryExpression
+}
+
+var _ sql.FunctionExpression = (*CurrentTimestamp)(nil)
+
+func NewCurrentTimestamp(val sql.Expression) sql.Expression {
+	return &CurrentTimestamp{expression.UnaryExpression{Child: val}}
+}
+
+func (c *CurrentTimestamp) FunctionName() string {
+	return "current_timestamp"
+}
+
+func (c *CurrentTimestamp) String() string {
+	return fmt.Sprintf("CURRENT_TIMESTAMP(%s)", c.Child.String())
+}
+
+func (c *CurrentTimestamp) Type() sql.Type { return sql.LongText }
+
+func (c *CurrentTimestamp) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 1 {
+		return nil, sql.ErrInvalidChildrenNumber.New(c, len(children), 1)
+	}
+	return NewCurrentTimestamp(children[0]), nil
+}
+
+func (c *CurrentTimestamp) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// Evaluate value
+	val, err := c.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	// If null, throw syntax error
+	if val == nil {
+		return nil, ErrTimeUnexpectedlyNil.New(c.FunctionName())
+	}
+
+	// Must receive integer, all other types throw syntax error
+	fsp := 0
+	switch val.(type) {
+	case int:
+		fsp = val.(int)
+	case int8:
+		fsp = int(val.(int8))
+	case int16:
+		fsp = int(val.(int16))
+	case int32:
+		fsp = int(val.(int32))
+	case int64:
+		fsp = int(val.(int64))
+	default:
+		return nil, ErrInvalidArgumentType.New(c.FunctionName())
+	}
+
+	// Parse and return answer
+	if fsp > maxCurrentTimestampPrecision {
+		return nil, ErrTooHighPrecision.New(fsp, c.FunctionName(), maxCurrentTimestampPrecision)
+	} else if fsp < 0 {
+		return nil, ErrInvalidArgumentType.New(c.FunctionName())
+	}
+
+	// Get the timestamp
+	t := ctx.QueryTime()
+
+	// Create non-nanosecond portion
+	timeString := fmt.Sprintf("%d-%02d-%02d %02d:%02d:%02d", t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second())
```

Comment:
Logic is fine for strings, but needs to return a time.Time truncated to the correct precision